framework/platform/bpl/include/bpl/bpl_cfg.h

```diff
@@ -91,6 +91,10 @@ namespace bpl {
 #define DEFAULT_BAND_STEERING 1
 #define DEFAULT_DFS_REENTRY 1
 #define DEFAULT_CLIENT_ROAMING 1
+#define DEFAULT_PERSISTENT_DB 0
+#define DEFAULT_CLIENTS_PERSISTENT_DB_MAX_SIZE 256
+#define DEFAULT_MAX_TIMELIFE_DELAY_DAYS 365
+#define DEFAULT_UNFRIENDLY_DEVICE_MAX_TIMELIFE_DELAY_DAYS 1
```

Comment:
These definitions should have been on the first and separate commit.
But again, I don't think that you should touch the values in case the function fails.
Either you don't fail at all, and then you can always change the output argument, or you make it possible to fail and then let the user handle it, which means that these should not be defined at all, at least not on the BPL.

Reply:
this should have been introduced before - my fixups commit hash must have been wrong - I will fix it (each API should introduce its own default.

there is a new approach to the BPL that user/platform should define (in UCI for example) only the values it must enforce for that platform, or values that differ from the different value which will be presented by a generic txt file but will actually be provided by the BPL.
this means the BPL returns a default value if the field is not defined in the UCI, instead of error.
my implementation follows that new approach. this is going to be changed for all BPL parameters soon.

Reply:
>this means the BPL returns a default value if the field is not defined in the UCI, **instead of error**.
If you return an error it is not the new approach then.

Reply:
there might be other errors in the process of retrieving the information.
note the difference between direct use of uci-get to the new API of -uci-get-default which returns a default value if the parameter doesn't exist in the UCI.
but there can be other errors happening which will result in an error and not in default value (from BPL perspective).